engine.py

```diff
@@ -912,3 +912,126 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self, session=None):
+        """
+        Returns the absolute path to the current file if it resides
+        on disk. If the session has never been saved and isn't
+        associated with a file on disk yet, an empty string is returned.
+
+        :param session: An object representing the active document
+                        (for MDI applications).
+        :returns: Path to the current file if it has been saved to disk,
+                  else returns an empty string.
+        :raises TankError: Raises a `TankError` if the application is
+                           unable to determine the session that is
+                           being referred to.
+        """
+        if self.hiero_enabled or self.studio_enabled:
+            import hiero
+
+            # first find what the current project is. Hiero is a multi
+            # project environment so we can ask the engine which
+            # project was last clicked.
+            selection = self.get_menu_selection()
+
+            # to guard against multiple selections, make sure we have
+            # only one selection.
+            if len(selection) != 1:
+                raise tank.TankError("Please select a single project.")
```

Comment:
i think the error message may be confusing for users - perhaps it should just emit a warning instead and say something like Warning: cannot determine the current session filename because multiple projects are currently selected?